src/ExceptionType.cc

```diff
@@ -78,6 +79,25 @@ tb_print_line_repeated(_PyUnicodeWriter *writer, long cnt)
 JSObject *ExceptionType::toJsError(JSContext *cx, PyObject *exceptionValue, PyObject *traceBack) {
   assert(exceptionValue != NULL);
 
+  // Gather JS context
+  JS_ReportErrorASCII(cx, ""); // throw JS error and gather all details
+
+  JS::ExceptionStack exceptionStack(cx);
+  if (!JS::GetPendingExceptionStack(cx, &exceptionStack)) {
+    return NULL;
+  }
+  JS_ClearPendingException(cx);
+
+  std::stringstream stackStream;
+  JS::HandleObject stackObj = exceptionStack.stack();
```

Comment:
you should never declare a variable as JS::Handle\<T\>. From the GC Rooting Guide: 

> "All GC thing pointers that are parameters to a function must be wrapped in JS::Handle\<T\>. A JS::Handle\<T\> is a reference to a JS::Rooted\<T\>, and is created implicitly by referencing a JS::Rooted\<T\>: **_It is not valid to create a JS::Handle\<T\> manually_** (the whole point of a Handle is that it only reference pointers that the GC knows about so it can update them when they move)."

It's curious that JS::ExceptionStack::stack returns a JS::HandleObject at all, as the GC Rooting Guide also says: 

> Common Pitfalls
> ...
> Returning a JS::Handle<T> from a function. If you do this, a handle may outlive its root.

so this is likely a bug in the Spidermonkey API.

```suggestion
  JS::RootedObject stackObj(cx, exceptionStack.stack();
```

Reply:
oh you're right, I've actually made fixes around this theme lately, and I blindly copied this over form somewhere else, so let me fix it there too